maps-compose/src/main/java/com/google/maps/android/compose/GoogleMap.kt

```diff
@@ -114,123 +113,152 @@ public fun GoogleMap(
-}
+/**
+ * Create and apply the [content] compositions to the map +
+ * dispose the [Composition] when the parent composable is disposed.
+ * */
+private fun CoroutineScope.launchSubcomposition(
+    mapUpdaterState: MapUpdaterState,
+    parentComposition: CompositionContext,
+    clickListeners: MapClickListeners,
+    content: (@Composable @GoogleMapComposable () -> Unit)?,
+    mapView: MapView
+): Job {
+    // Use [CoroutineStart.UNDISPATCHED] to kick off GoogleMap loading immediately
+    return launch(start = CoroutineStart.UNDISPATCHED) {
+        val map = mapView.awaitMap()
+        val composition = Composition(
+            applier = MapApplier(map, mapView),
+            parent = parentComposition
+        )
 
-private suspend inline fun MapView.newComposition(
-    parent: CompositionContext,
-    mapClickListeners: MapClickListeners,
-    noinline content: @Composable () -> Unit
-): Composition {
-    val map = awaitMap()
-    return Composition(
-        MapApplier(map, this, mapClickListeners), parent
-    ).apply {
-        setContent(content)
-    }
-}
+        try {
+            composition.setContent {
+                MapUpdater(mapUpdaterState)
 
-/**
- * Registers lifecycle observers to the local [MapView].
- */
-@Composable
-private fun MapLifecycle(mapView: MapView) {
-    val context = LocalContext.current
-    val lifecycle = LocalLifecycleOwner.current.lifecycle
-    val previousState = remember { mutableStateOf(Lifecycle.Event.ON_CREATE) }
-    DisposableEffect(context, lifecycle, mapView) {
-        val mapLifecycleObserver = mapView.lifecycleObserver(previousState)
-        val callbacks = mapView.componentCallbacks()
-
-        lifecycle.addObserver(mapLifecycleObserver)
-        context.registerComponentCallbacks(callbacks)
-
-        onDispose {
-            lifecycle.removeObserver(mapLifecycleObserver)
-            context.unregisterComponentCallbacks(callbacks)
-        }
-    }
-    DisposableEffect(mapView) {
-        onDispose {
-            mapView.onDestroy()
-            mapView.removeAllViews()
-        }
-    }
-}
+                MapClickListenerUpdater(clickListeners)
 
-private fun MapView.lifecycleObserver(previousState: MutableState<Lifecycle.Event>): LifecycleEventObserver =
-    LifecycleEventObserver { _, event ->
-        event.targetState
-        when (event) {
-            Lifecycle.Event.ON_CREATE -> {
-                // Skip calling mapView.onCreate if the lifecycle did not go through onDestroy - in
-                // this case the GoogleMap composable also doesn't leave the composition. So,
-                // recreating the map does not restore state properly which must be avoided.
-                if (previousState.value != Lifecycle.Event.ON_STOP) {
-                    this.onCreate(Bundle())
+                CompositionLocalProvider(
+                    LocalCameraPositionState provides currentCameraPositionState
+                ) {
+                    content?.invoke()
```

Comment:
`currentCameraPositionState` is not valid here. After the `content` type change, this whole expression can change to simply:

```
CompositionLocalProvider(
    LocalCameraPositionState provides mapUpdaterState.cameraPositionState,
    content
)
```

Reply:
Fixed 👍